### DISCUSSION TITLE
Thanos Ruler querying Thanos Query Frontend instead of Thanos Querier

### DISCUSSION THREAD

**Discussion by jaanhio** [thanos-io/thanos#4152] [Questions & Answers]
I am currently facing an issue where Thanos Querier is hitting a performance bottleneck due to the amount of queries from Thanos Ruler. 

In this case, is it a good idea to have Thanos Ruler query Thanos Query Frontend instead? my concern is the querying of stale metrics for alerting but not 100% sure if that will be an issue.

Look forward to advice on this!

**Comment by yeya24** [comment]
Hi @jaanhio, it doesn't help IMO. Thanos Query Frontend works well for Range Query. However, the Rules related queries are simply instant queries that cannot be split and cached by Query Frontend. So I think there is no performance improvement of using it.
